### DISCUSSION TITLE
Searching leaves of NFTs including image_data too slow

### DISCUSSION THREAD

**Discussion by youngzhenhao** [lightninglabs/taproot-assets#1266] [Q&A]
I minted and finalised some collectible type taproot assets on the bitcoin network and followed this [format](https://github.com/snow884/taproot-assets-metadata-format?tab=readme-ov-file#fields), so I included image_data in the metadata.
After I minted 300 NFTs, searching the tweaked_group_key leaves in the universe became very slow, but I have thousands of NFTs to mint.
Is it true that I can only include urls instead of image_data in thousands of NFTs?
What is the best way to do this?

**Comment by guggero** [comment]
> searching the tweaked_group_key leaves in the universe became

Can you tell us which URI you mean? And with slow, you mean the response time or the time to process the response? And against you own universe or a default one?
Perhaps some caching or database optimization can help here.
